From 94fc579ada157b0bc971bc2d8f47e15d3b8663eb Mon Sep 17 00:00:00 2001
From: Kemal Rasim Sh <quic_kshakir@quicinc.com>
Date: Mon, 24 Mar 2025 15:14:50 +0200
Subject: [PATCH 1/3] Delete pipeline even if state update to NULL returns
 error

If a plugin internally encounters an unrecoverable error,
causes it to always return error on state change
then the GSTD cannot discard the pipeline.

Upstream-Status: Inappropriate

Signed-off-by: Kemal Rasim Sh <quic_kshakir@quicinc.com>
---
 libgstd/gstd_pipeline_deleter.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libgstd/gstd_pipeline_deleter.c b/libgstd/gstd_pipeline_deleter.c
index d90931d..3817bd1 100644
--- a/libgstd/gstd_pipeline_deleter.c
+++ b/libgstd/gstd_pipeline_deleter.c
@@ -97,9 +97,8 @@ gstd_pipeline_deleter_delete (GstdIDeleter * iface, GstdObject * object)
   if (ret)
     return ret;
 
-  ret = gstd_object_update (state, "NULL");
-  if (ret)
-    return ret;
+  /* Delete the object even it fails */
+  gstd_object_update (state, "NULL");
 
   g_object_unref (state);
   g_object_unref (object);
-- 
2.25.1

